### ISSUE TITLE
babel fails to install under npm@3.1.0 on windows

### ISSUE THREAD

**Issue by jamietre** [npm/npm#8878]
Works fine under 2.11.2, I assume this is an npm@3 issue.

    { }  » npm install -g babel                                                                                    

    C:\Users\_\AppData\Roaming\npm\babel-plugin -> C:\Users\_\AppData\Roaming\npm\node_modules\babel\bin\babel-plugin\index.js
    C:\Users\_\AppData\Roaming\npm\babel-node -> C:\Users\_\AppData\Roaming\npm\node_modules\babel\bin\babel-node
    C:\Users\_\AppData\Roaming\npm\babel -> C:\Users\_\AppData\Roaming\npm\node_modules\babel\bin\babel\index.js
    C:\Users\_\AppData\Roaming\npm\babel-external-helpers -> C:\Users\_\AppData\Roaming\npm\node_modules\babel\bin\babel-external-helpers

    > fsevents@0.3.6 install C:\Users\_\AppData\Roaming\npm\node_modules\babel\node_modules\fsevents
    > node-gyp rebuild

    C:\Users\_\AppData\Roaming\npm\node_modules\babel\node_modules\fsevents>if not defined npm_config_node_gyp (node "C:\Users\_\AppData\Roaming\npm\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node  rebuild )
    Traceback (most recent call last):
      File "C:\Users\_\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\gyp\gyp_main.py", line 18, in <module>
        sys.exit(gyp.script_main())
      File "C:\Users\_\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\__init__.py", line 533, in script_main
        return main(sys.argv[1:])
      File "C:\Users\_\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\__init__.py", line 526, in main
        return gyp_main(args)
      File "C:\Users\_\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\__init__.py", line 511, in gyp_main
        generator.GenerateOutput(flat_list, targets, data, params)
      File "C:\Users\_\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\generator\msvs.py", line 2000, in GenerateOutput
        sln_projects, project_objects, flat=msvs_version.FlatSolution())
      File "C:\Users\_\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\generator\msvs.py", line 1759, in _GatherSolutionFolders
        return _DictsToFolders('', root, flat)
      File "C:\Users\_\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\generator\msvs.py", line 1712, in _DictsToFolders
        for folder, contents in bucket.iteritems():
    AttributeError: 'MSVSProject' object has no attribute 'iteritems'
    gyp ERR! configure error
    gyp ERR! stack Error: `gyp` failed with exit code: 1
    gyp ERR! stack     at ChildProcess.onCpExit (C:\Users\_\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\configure.js:355:16)
    gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
    gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1074:12)
    gyp ERR! System Windows_NT 6.1.7601
    gyp ERR! command "node" "C:\\Users\\_\\AppData\\Roaming\\npm\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
    gyp ERR! cwd C:\Users\_\AppData\Roaming\npm\node_modules\babel\node_modules\fsevents
    gyp ERR! node -v v0.12.6
    gyp ERR! node-gyp -v v2.0.1
    gyp ERR! not ok
    C:\Users\_\AppData\Roaming\npm
    └── (empty)

    npm WARN EPACKAGEJSON C:\Users\_\AppData\Roaming\npm No description
    npm WARN EPACKAGEJSON C:\Users\_\AppData\Roaming\npm No repository field.
    npm WARN EPACKAGEJSON C:\Users\_\AppData\Roaming\npm No README data
    npm WARN EPACKAGEJSON C:\Users\_\AppData\Roaming\npm No license field.
    npm ERR! Windows_NT 6.1.7601
    npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\users\\_\\appdata\\roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "install" "-g" "babel"
    npm ERR! node v0.12.6
    npm ERR! npm  v3.1.0
    npm ERR! code ELIFECYCLE

    npm ERR! fsevents@0.3.6 install: `node-gyp rebuild`
    npm ERR! Exit status 1
    npm ERR!
    npm ERR! Failed at the fsevents@0.3.6 install script 'node-gyp rebuild'.
    npm ERR! This is most likely a problem with the fsevents package,
    npm ERR! not with npm itself.
    npm ERR! Tell the author that this fails on your system:
    npm ERR!     node-gyp rebuild
    npm ERR! You can get their info via:
    npm ERR!     npm owner ls fsevents
    npm ERR! There is likely additional logging output above.

    npm ERR! Please include the following file with any support request:
    npm ERR!     D:\code\NpmTest\Library\utility\npm-debug.log
    npm ERR! code 1

**Comment by radum** [comment]
Same issue here although the error is a bit different. Works with 2.x npm but fails on 3.x.

I've installed `https://github.com/TooTallNate/node-gyp` as per the instructions and other modules that require a build are working fine, so the config is done Ok I guess for `node-gyp` to work properly.

I'm on a Win x64 machine, node 0.12.7, npm 3.1.1

```
$ npm install -g babel
C:\Users\zamolxe\AppData\Roaming\npm\babel-external-helpers -> C:\Users\zamolxe\AppData\Roaming\npm\node_modules\babel\bin\babel-external-helpers
C:\Users\zamolxe\AppData\Roaming\npm\babel-node -> C:\Users\zamolxe\AppData\Roaming\npm\node_modules\babel\bin\babel-node
C:\Users\zamolxe\AppData\Roaming\npm\babel -> C:\Users\zamolxe\AppData\Roaming\npm\node_modules\babel\bin\babel\index.js
C:\Users\zamolxe\AppData\Roaming\npm\babel-plugin -> C:\Users\zamolxe\AppData\Roaming\npm\node_modules\babel\bin\babel-plugin\index.js

> fsevents@0.3.6 install C:\Users\zamolxe\AppData\Roaming\npm\node_modules\babel\node_modules\fsevents
> node-gyp rebuild


C:\Users\zamolxe\AppData\Roaming\npm\node_modules\babel\node_modules\fsevents>if not defined npm_config_node_gyp (node "c:\Users\zamolxe\AppData\Roaming\npm\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node  rebuild )
gyp WARN install got an error, rolling back install
gyp ERR! configure error
gyp ERR! stack Error: 404 response dowloading http://nodejs.org/dist/v2.3.2/node-v2.3.2.tar.gz
gyp ERR! stack     at Request.<anonymous> (c:\Users\zamolxe\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\install.js:251:14)
gyp ERR! stack     at emitOne (events.js:82:20)
gyp ERR! stack     at Request.emit (events.js:169:7)
gyp ERR! stack     at Request.onRequestResponse (c:\Users\zamolxe\AppData\Roaming\npm\node_modules\npm\node_modules\request\request.js:985:10)
gyp ERR! stack     at emitOne (events.js:77:13)
gyp ERR! stack     at ClientRequest.emit (events.js:169:7)
gyp ERR! stack     at HTTPParser.parserOnIncomingClient (_http_client.js:415:21)
gyp ERR! stack     at HTTPParser.parserOnHeadersComplete (_http_common.js:88:23)
gyp ERR! stack     at Socket.socketOnData (_http_client.js:305:20)
gyp ERR! stack     at emitOne (events.js:77:13)
gyp ERR! System Windows_NT 6.1.7601
gyp ERR! command "c:\\Program Files\\iojs\\node.exe" "c:\\Users\\zamolxe\\AppData\\Roaming\\npm\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\zamolxe\AppData\Roaming\npm\node_modules\babel\node_modules\fsevents
gyp ERR! node -v v2.3.2
gyp ERR! node-gyp -v v2.0.1
gyp ERR! not ok
npm WARN EPACKAGEJSON C:\Users\zamolxe\AppData\Roaming\npm No description
npm WARN EPACKAGEJSON C:\Users\zamolxe\AppData\Roaming\npm No repository field.
npm WARN EPACKAGEJSON C:\Users\zamolxe\AppData\Roaming\npm No README data
npm WARN EPACKAGEJSON C:\Users\zamolxe\AppData\Roaming\npm No license field.
npm ERR! Windows_NT 6.1.7601
npm ERR! argv "c:\\Program Files\\iojs\\node.exe" "c:\\Users\\zamolxe\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "install" "-g" "babel"
npm ERR! node v2.3.2
npm ERR! npm  v3.1.1
npm ERR! code ELIFECYCLE

npm ERR! fsevents@0.3.6 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the fsevents@0.3.6 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the fsevents package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls fsevents
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     d:\MyProjects\work-dev\onlineops-survey-template\npm-debug.log
```

**Comment by cchamberlain** [comment]
Getting this with lots of packages, I added my details to https://github.com/TooTallNate/node-gyp/issues/661.

**Comment by timdp** [comment]
As pointed out by @rchanou in TooTallNate/node-gyp#661, this actually seems to be caused by npm/npm#8921. Please watch that one for updates.

**Comment by iarna** [comment]
Hey! Good news, this should be fixed in the just released `npm@3.2.0`!

**Comment by jamietre** [comment]
Confirmed, seems to be working well! Thanks!
